Improbable/Schema/Improbable.Schema.Bundle/FieldDefinitionExtensions.cs

```diff
@@ -0,0 +1,296 @@
+using System;
+
+namespace Improbable.Schema.Bundle
+{
+    public static class FieldDefinitionExtensions
+    {
+        public static bool IsEnum(this FieldDefinition field)
+        {
+            return field.TypeSelector switch
+            {
+                FieldType.Option => field.OptionType.InnerType.ValueTypeSelector == ValueType.Enum,
+                FieldType.List => field.ListType.InnerType.ValueTypeSelector == ValueType.Enum,
+                FieldType.Map => throw new InvalidOperationException(),
+                FieldType.Singular => field.SingularType.Type.ValueTypeSelector == ValueType.Enum,
+                _ => throw new ArgumentOutOfRangeException()
+            };
+        }
+
+        public static bool IsPrimitive(this FieldDefinition field)
+        {
+            return field.TypeSelector switch
+            {
+                FieldType.Option => field.OptionType.InnerType.ValueTypeSelector == ValueType.Primitive,
+                FieldType.List => field.ListType.InnerType.ValueTypeSelector == ValueType.Primitive,
+                FieldType.Map => throw new InvalidOperationException(),
+                FieldType.Singular => field.SingularType.Type.ValueTypeSelector == ValueType.Primitive,
+                _ => throw new ArgumentOutOfRangeException()
```

Comment:
What's the rationale for checking the inner types for option & list, but not map? This seems a little odd to me.

Reply:
I...don't know why I left that out. Will add it in.

Reply:
Ah yes, I know, it's kind of nonsensical since maps have two types. I've added a message to the exception to check the Key/Value `TypeReference`s of maps individually.